rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/leveldb/leveldb_wrapper.hpp

```diff
@@ -0,0 +1,172 @@
+// Copyright 2020 Sony Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_STORAGE_DEFAULT_PLUGINS__LEVELDB__LEVELDB_WRAPPER_HPP_
+#define ROSBAG2_STORAGE_DEFAULT_PLUGINS__LEVELDB__LEVELDB_WRAPPER_HPP_
+
+#include <leveldb/db.h>
+#include <leveldb/write_batch.h>
+#include <leveldb/comparator.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <utility>
+#include <memory>
+
+#include "rosbag2_storage/metadata_io.hpp"
+#include "rosbag2_storage/serialized_bag_message.hpp"
+#include "rosbag2_storage_default_plugins/leveldb/leveldb_exception.hpp"
+#include "rosbag2_storage_default_plugins/visibility_control.hpp"
+
+namespace rosbag2_storage_plugins
+{
+
+#define swap_64(x) \
+  ((((x) & 0xff00000000000000ull) >> 56) \
+  | (((x) & 0x00ff000000000000ull) >> 40) \
+  | (((x) & 0x0000ff0000000000ull) >> 24) \
+  | (((x) & 0x000000ff00000000ull) >> 8) \
+  | (((x) & 0x00000000ff000000ull) << 8) \
+  | (((x) & 0x0000000000ff0000ull) << 24) \
+  | (((x) & 0x000000000000ff00ull) << 40) \
+  | (((x) & 0x00000000000000ffull) << 56))
+
+class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC LeveldbWrapper
```

Comment:
having this wrapper to absorb the difference of database file structure against sqlite3. just a comment to confirm that we have the same understanding.

Reply:
Yeah. This wrapper wrap the detailed operation (open, read, etc) of leveldb.